packages/editor/src/components/post-schedule/index.js

```diff
@@ -2,14 +2,65 @@
  * WordPress dependencies
  */
 import { __experimentalGetSettings } from '@wordpress/date';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { DateTimePicker } from '@wordpress/components';
-import { useRef } from '@wordpress/element';
+import { useRef, useState, useMemo } from '@wordpress/element';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editorStore } from '../../store';
+
+function getDayOfTheMonth( date = new Date(), firstDay = true ) {
+	const d = new Date( date );
+	return new Date(
+		d.getFullYear(),
+		d.getMonth() + ( firstDay ? 0 : 1 ),
+		firstDay ? 1 : 0
+	).toISOString();
+}
+
+export default function PostSchedule() {
+	const date = useSelect(
+		( select ) => select( editorStore ).getEditedPostAttribute( 'date' ),
+		[]
+	);
+
+	const { editPost } = useDispatch( editorStore );
+	const onUpdateDate = ( postDate ) => editPost( { date: postDate } );
+
+	const [ previewedMonth, setPreviewedMonth ] = useState(
+		getDayOfTheMonth( date )
+	);
+
+	// Pick up published and schduled site posts.
+	const postsEvents = useSelect(
+		( select ) =>
+			select( coreStore ).getEntityRecords( 'postType', 'post', {
```

Comment:
We either want this to get all post types (which seems a bit risky) or search for records with the currently edited post type. 

Should we add a filter on these events too, so plugins etc, can manipulate what is displayed?

Reply:
In case we'd like to add a filter, what should be the filter name? `editor.calendarEvents` ?

Reply:
> ...or search for records with the currently edited post type.

It sounds reasonable to me. Addressed here https://github.com/WordPress/gutenberg/pull/29716/commits/383892d4ac51b7b5e5ef8b586ff7cb6e642aa8c7
We can get rid of it in case we want to.

Reply:
> Should we add a filter on these events too, so plugins etc, can manipulate what is displayed?

Addressed here https://github.com/WordPress/gutenberg/pull/29716/commits/8403efd07d5435e891ea52e19b5b9e7e31169ae8

I tested with my dev env and it works very well. The following example extends the events adding an event for _today_.

```es6
function addCustomCalendarEvents( DateTimePickerComponent ) {
	return ( props ) => (
		<DateTimePickerComponent
			{ ...props }
			events={ [
				...props.events,
				{
					title: 'The future is now!',
					date: new Date(),
				},
			] }
		/>
	);
}

addFilter( 'editor.CalendarEvents', 'my-plugin/calendar-events', addCustomCalendarEvents );
```

<img src="https://user-images.githubusercontent.com/77539/111488225-d0158180-8717-11eb-9e60-67b2e365b54e.png" width="300px" />

Reply:
Looks good! I'm not sure what the filter name should be, but `editor.CalendarEvents` seems reasonable.

Reply:
Me neither honestly. I took a look at the [editor filters](https://developer.wordpress.org/block-editor/reference-guides/filters/editor-filters/) to try to follow the convention.

Reply:
if we agree to add this filter, I can update the doc.

Reply:
Updated the doc here https://github.com/WordPress/gutenberg/pull/29716/commits/e4454ec89a5122c4baa2ab7d7c0cf5cf7efc9c21